classes/plain.lua

```diff
@@ -310,10 +310,14 @@ function class:registerCommands ()
     SILE.settings:set("linebreak.tolerance", 10000)
   end)
 
-  self:registerCommand("center", function (_, content)
+  self:registerCommand("center", function (options, content)
     if #SILE.typesetter.state.nodes ~= 0 then
       SU.warn("\\center environment started after other nodes in a paragraph, may not center as expected")
     end
+    if SU.boolean(options.vertical) then
+      SILE.call("hbox")
+      SILE.call("vfill")
```

Comment:
A vfill won't push the content middle of the page, _unless_ if it is a final page automatically vfill'ed (such as the final page of your example) or explicitly `\eject`'ed. Then another vfill is currently automatically added at the end of the page, resulting in centering.

Personally:
- I don't think that horizontal centering and vertical centering should be the same command
- I don' think the vfill added on final pages or via eject is actually good :)